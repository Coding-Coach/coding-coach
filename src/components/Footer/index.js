import React, { Fragment } from 'react';
import { Link } from 'gatsby'

import CookiesBanner from '../Banner/Cookies';
import ScrollTop from '../Button/ScrollTop';

export default function Footer() {
  return (
    <Fragment>
      <footer className="container">
        <div className="pt-4 pb-8 border-t border-secondary-lightest clearfix">
          <ul className="float-right">
            <LinkItem to="/legal/terms-conditions">Terms & Conditions</LinkItem>
            <LinkItem to="/legal/cookies">Cookies</LinkItem>
            <LinkItem to="/legal/privacy">Privacy Policy</LinkItem>
            <LinkItem to="/legal/code-of-conduct">Code Of Conduct</LinkItem>
          </ul>
        </div>
      </footer>
      <CookiesBanner />
      <ScrollTop />
    </Fragment>
  );
}

function LinkItem({ children, to }) {
  return (
    <li className="mr-4 inline-block md:ml-4 md:mr-0">
      <Link
        to={to}
        className="text-body text-secondary-dark hover:underline hover:text-primary-light"
      >
        {children}
      </Link>
    </li>
  );
}
