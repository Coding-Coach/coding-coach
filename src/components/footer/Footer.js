import React from 'react';
import { translate } from 'react-i18next';
import 'i18n/i18n';

// @TODO: Bring it back when ready
// import DonateButton from './DonateButton';
import styles from './footer.scss';

function Footer({ onClickLegal, t }) {
  return (
    <footer className={styles.footerMain}>
      <div className={styles.footerInner}>
        <div className={styles.column}>{/* <DonateButton /> */}</div>

        <a
          onClick={() => onClickLegal('terms-and-conditions')}
          href={'#terms'}
          className={styles.footerLink}
        >
          {t('legal-terms-and-conditions')}
        </a>
        <a onClick={() => onClickLegal('cookies')} href={'#cookies'} className={styles.footerLink}>
          {t('legal-cookies')}
        </a>
        <a
          onClick={() => onClickLegal('privacy-policy')}
          href={'#privacy'}
          className={styles.footerLink}
        >
          {t('legal-privacy-policy')}
        </a>
      </div>
    </footer>
  );
}

export default translate('translations')(Footer);
