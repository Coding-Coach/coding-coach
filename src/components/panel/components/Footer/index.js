import PanelFooter from './PanelFooter';

export default PanelFooter;
