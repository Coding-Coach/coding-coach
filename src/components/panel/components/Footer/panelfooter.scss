
.footer {
  display: flex;

  &::first-child {
    margin-left: auto;
  }
}

.pullRight {
  margin-left: auto;
}