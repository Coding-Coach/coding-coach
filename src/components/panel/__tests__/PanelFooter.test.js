import React from 'react';
import { render } from 'react-testing-library';
import Footer from '../PanelFooter';

test('renders the PanelFooter component', () => {
  const { container } = render(<Footer />);

  expect(container.firstChild).toMatchSnapshot();

  const footer = container.querySelectorAll('footer');
  expect(footer.length).toEqual(1);
});
