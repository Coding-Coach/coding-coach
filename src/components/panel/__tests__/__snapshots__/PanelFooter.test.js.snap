// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the PanelFooter component 1`] = `
<footer
  class="footer"
/>
`;
